docs/advanced/checked-build.md

```diff
@@ -0,0 +1,60 @@
+---
+title: Checked Builds
+redirect_from:
+  - /Checked_Build/
+---
+
+Introduction
+============
+
+The Mono VM uses asserts throughout to ensure it is in a correct state. There are a few classes of asserts which are useful for debugging Mono, but which are too expensive to include in a production VM. If you are doing development on Mono, you can enable these with a specially built VM.
+
+To enable the checked modes, you must build with the `--with-checked-build=` argument passed to `autogen.sh`/`configure`, and then run with the `MONO_CHECK_MODE` environment variable. Both `--with-checked-build` and `MONO_CHECK_MODE` should be set to either the word `all`, or a comma-separated list of mode names if you need to do fine-grained testing. (In particular you might want to skip `metadata` mode, since it is exceptionally slow.)
```

Comment:
Why do I need to set both the configure flag and the env var to the same list? What happens if they don't match?